core/src/main/java/hudson/cli/DeleteBuildsCommand.java

```diff
@@ -54,10 +55,16 @@ protected void printUsageSummary(PrintStream stderr) {
     protected int act(List<AbstractBuild<?, ?>> builds) throws IOException {
         job.checkPermission(Run.DELETE);
 
-        for (AbstractBuild build : builds)
-            build.delete();
+        final HashSet<Integer> hs_builds = new HashSet<Integer>();
 
-        stdout.println("Deleted "+builds.size()+" builds");
+        for (AbstractBuild build : builds) {
+            if (!hs_builds.contains(build.number)) {
+                build.delete();
+                hs_builds.add(build.number);
+            }
+        }
```

Comment:
Not sure I understand what you're trying to do here.
Reading the JIRA, I understand that an `IOException` can be thrown if we try to remove an already removed build. Shouldn't you just change the original code with a try/catch?

Or possibly improve/throw a subclass of IOException in https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/Run.java#L1480 to be better informed the to-be-removed build seems actually to be (already) absent?